File: legacy-python/envx/cli/commands/list.py
"""List command."""

from __future__ import annotations

from typing import Annotated

import typer

from envx.cli.commands.common import build_service, handle_error, render_rows
from envx.core.exceptions import EnvxError


def register(app: typer.Typer) -> None:
    """Register the list command."""

    @app.command(name="list", help="List environment variables.")
    def list_command(
        env: str | None = typer.Option(
            None, "--env", help="Target environment. Defaults to the active environment."
        ),
        show_secrets: bool = typer.Option(
            False, "--show-secrets", help="Decrypt and display secret values."
        ),
        output_format: Annotated[
            str,
            typer.Option(
                "--format",
                help="Output format.",
                case_sensitive=False,
            ),
        ] = "table",
    ) -> None:
        service = build_service()
        try:
            rows = service.list_variables(env_name=env, show_secrets=show_secrets)
        except EnvxError as exc:
            handle_error(exc)
        render_rows(rows, output_format.lower())

File: internal/cli/list.go
package cli

import (
	"errors"
	"fmt"
	"io"
)

func cmdList(args []string, stdout, stderr io.Writer) int {
	root, rest := splitRootFlag(args)
	if checkHelp(rest, stdout, "list") {
		return 0
	}
	envName := ""
	showSecrets := false
	format := "table"
	for len(rest) > 0 {
		switch rest[0] {
		case "--env":
			if len(rest) < 2 {
				return printError(stderr, errors.New("missing value for '--env'"))
			}
			envName = rest[1]
			rest = rest[2:]
		case "--show-secrets":
			showSecrets = true
			rest = rest[1:]
		case "--format":
			if len(rest) < 2 {
				return printError(stderr, errors.New("missing value for '--format'"))
			}
			format = rest[1]
			rest = rest[2:]
		default:
			return printError(stderr, fmt.Errorf("unknown option '%s'", rest[0]))
		}
	}

	service := projectService(root)
	rows, err := service.ListVariables(envName, showSecrets)
	if err != nil {
		return printError(stderr, err)
	}
	renderVariableRows(rows, format, stdout)
	return 0
}
